src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java

```diff
@@ -0,0 +1,161 @@
+    private static final String CASE_DATA_PREFIX = "data.";
+    private static final String COLLECTION_VALUE_SUFFIX = ".value";
+    private static final String KEYWORD_SUFFIX = ".keyword";
+
+    private final ObjectMapper objectMapper;
+    private final ApplicationParams applicationParams;
+    private final ObjectMapperService objectMapperService;
+    private final SearchQueryOperation searchQueryOperation;
+    private final GetCaseTypeOperation getCaseTypeOperation;
+    private final ElasticsearchMappings elasticsearchMappings;
+
+    @Autowired
+    public ElasticsearchQueryHelper(@Qualifier("DefaultObjectMapper") ObjectMapper objectMapper,
+                                    ApplicationParams applicationParams,
+                                    ObjectMapperService objectMapperService,
+                                    SearchQueryOperation searchQueryOperation,
+                                    @Qualifier(AuthorisedGetCaseTypeOperation.QUALIFIER) GetCaseTypeOperation getCaseTypeOperation,
+                                    ElasticsearchMappings elasticsearchMappings) {
+        this.objectMapper = objectMapper;
+        this.applicationParams = applicationParams;
+        this.objectMapperService = objectMapperService;
+        this.searchQueryOperation = searchQueryOperation;
+        this.getCaseTypeOperation = getCaseTypeOperation;
+        this.elasticsearchMappings = elasticsearchMappings;
+    }
+
+    public CrossCaseTypeSearchRequest prepareRequest(List<String> caseTypeIds, String jsonSearchRequest) {
+        if (CollectionUtils.isEmpty(caseTypeIds)) {
+            throw new BadSearchRequest("At least one case type ID is required.");
+        }
+
+        JsonNode searchRequest = validateAndConvertRequest(jsonSearchRequest);
+
+        return new CrossCaseTypeSearchRequest.Builder()
+            .withCaseTypes(caseTypeIds)
+            .withSearchRequest(searchRequest)
+            .build();
+    }
+
+    public CrossCaseTypeSearchRequest prepareRequest(String caseTypeId, String jsonSearchRequest, String useCase) {
+        if (Strings.isNullOrEmpty(caseTypeId)) {
+            throw new BadSearchRequest("Case type ID is required.");
+        }
+
+        JsonNode searchRequest = validateAndConvertRequest(jsonSearchRequest);
+        applyConfiguredSort(searchRequest, caseTypeId, useCase);
+
+        return new CrossCaseTypeSearchRequest.Builder()
+            .withCaseTypes(Collections.singletonList(caseTypeId))
+            .withSearchRequest(searchRequest)
+            .build();
+    }
+
+    private void applyConfiguredSort(JsonNode searchRequest, String caseTypeId, String useCase) {
+        JsonNode sortNode = searchRequest.get(SORT);
+        if (sortNode == null || useCase == null) {
+            ArrayNode appliedSortsNode = buildSortNode(caseTypeId, useCase);
+            if (appliedSortsNode.size() > 0) {
+                ((ObjectNode)searchRequest).set(SORT, appliedSortsNode);
+            }
+        }
```

Comment:
I would strongly suggest we introduce a separated class that deals only with manipulation of the JSON response. Nothing else. 
For example a ElasticSearchRequest class. With methods like:
new ElasticSearchRequest(jsonRequest)
esr.toJson()
esr.isSorted()
esr.addSorting(fields)
For now we can use it here only and then pass down to the external API the json request by calling .toJson(). So we don't have to change much code